src/pages/Account/NewCardForm.tsx

```diff
@@ -0,0 +1,294 @@
+import * as React from 'react';
+import { Button, Form, Row, Col, FormGroup, Label, FormFeedback, Input } from 'reactstrap';
+import { GET_PAYMENT_SOURCES, CREATE_PAYMENT_SOURCE } from 'networking/paymentSources';
+import { compose, graphql } from 'react-apollo';
+import { injectStripe, CardNumberElement, CardExpiryElement, CardCVCElement, PostalCodeElement } from 'react-stripe-elements';
+import StripeWrapper from 'HOCs/StripeWrapper';
+import styled from 'styled-components';
+
+interface Props {
+  stripe: Window['Stripe'];
+  createPaymentSource: (stripeToken: string) => void;
+  handleModal: (modal?: string) => void;
+  setAlert: (msg?: string) => void;
+}
+
+interface State {
+  completed: {
+    cardExpiry: boolean;
+    cardNumber: boolean;
+    cardCvc: boolean;
+    name: boolean;
+    postalCode: boolean;
+    [key: string]: boolean;
+  },
+  errors: {
+    cardExpiry: string;
+    cardNumber: string;
+    cardCvc: string;
+    name: string;
+    postalCode: string;
+    submit: string;
+    [key: string]: string;
```

Comment:
For `completed` and `errors`, do we lose anything useful if we just say `[key: string]: boolean` and `[key: string]: string` and omit the fields? (Even better, can these be `[key: NewCardField]`? I feel like we found out the answer was *no* last time I wondered that...)

More broadly, from here to where state is initialized we list out our form properties five times, and that's all just boilerplate. Would be great to figure out how to get that down to a single schema/interface/enum/something here, with one of our smaller forms, and hopefully come out with an approach we can use elsewhere

Reply:
removed the explicit type declarations.
> Even better, can these be `[key: NewCardField]`?
ya typescript doesn't like that =(

Would love to reduce the boilerplate! Will keep thinking of better solutions and am open to suggestions. This form may be a special case since Stripe Elements abstract a lot of the implementation and Formik isn't used.